androidApp/src/main/java/com/mbta/tid/mbta_app/android/favorites/FavoritesViewModel.kt

```diff
@@ -0,0 +1,89 @@
+package com.mbta.tid.mbta_app.android.favorites
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.mbta.tid.mbta_app.dependencyInjection.UsecaseDI
+import com.mbta.tid.mbta_app.model.RouteCardData
+import com.mbta.tid.mbta_app.model.RouteStopDirection
+import com.mbta.tid.mbta_app.model.response.AlertsStreamDataResponse
+import com.mbta.tid.mbta_app.model.response.GlobalResponse
+import com.mbta.tid.mbta_app.model.response.PredictionsStreamDataResponse
+import com.mbta.tid.mbta_app.model.response.ScheduleResponse
+import com.mbta.tid.mbta_app.usecases.FavoritesUsecases
+import io.github.dellisd.spatialk.geojson.Position
+import kotlinx.datetime.Instant
+
+class FavoritesViewModel(
+    private val favoritesUsecases: FavoritesUsecases = UsecaseDI().favoritesUsecases
+) : ViewModel() {
+    var favorites by mutableStateOf<Set<RouteStopDirection>?>(null)
+    var routeCardData by mutableStateOf<List<RouteCardData>?>(null)
+
+    suspend fun loadFavorites() {
+        favorites = favoritesUsecases.getRouteStopDirectionFavorites()
+    }
+
+    suspend fun loadRouteCardData(
+        global: GlobalResponse?,
+        location: Position?,
+        schedules: ScheduleResponse?,
+        predictions: PredictionsStreamDataResponse?,
+        alerts: AlertsStreamDataResponse?,
+        now: Instant,
+    ) {
+        val stops = favorites?.map { global?.getStop(it.stop) }.orEmpty()
+        val stopIds = stops.mapNotNull { it?.childStopIds?.plus(listOf(it.id)) }.flatten()
+        if (stopIds.isEmpty()) {
+            routeCardData = emptyList()
+            return
+        }
+        if (global == null || location == null) {
+            return
+        }
+        val loadedRouteCardData =
+            RouteCardData.routeCardsForStopList(
+                stopIds,
+                global,
+                location,
+                schedules,
+                predictions,
+                alerts,
+                now,
+                emptySet(),
+                RouteCardData.Context.Favorites,
+            )
+        routeCardData = filterRouteAndDirection(loadedRouteCardData, global)
+    }
+
+    fun filterRouteAndDirection(
+        routeCardData: List<RouteCardData>?,
+        global: GlobalResponse,
+    ): List<RouteCardData>? {
+        return routeCardData?.filter { data ->
+            data.stopData =
```

Comment:
If you’re returning a new list anyway, it seems like this wouldn’t have to be an in-place mutation, and you could `.copy()` instead to avoid having to make `stopData` a `var`.